src/sagemaker_xgboost_container/data_utils.py

```diff
@@ -43,13 +43,49 @@ def _get_invalid_csv_error_msg(line_snippet, file_name):
     return INVALID_CONTENT_FORMAT_ERROR.format(line_snippet=line_snippet, file_name=file_name, content_type='CSV')
 
 
+def _get_csv_content_type(content_type_cfg_val):
+    """
+    Returns CSV if content_type_cfg_val is
+    * 'csv',
+    * 'text/csv',
+    * 'text/csv; ...' with valid parameters
+    """
+    if content_type_cfg_val in [CSV, _content_types.CSV]:
+        # Allow 'csv' and 'text/csv'
+        return CSV
+    elif _content_types.CSV in content_type_cfg_val:
```

Comment:
This seems like a really ugly way to deal with a general problem. Have you looked into existing tools that can parse request header content types? such as: https://pypi.org/project/python-mimeparse/

It just feels like we are trying to reinvent the wheel here.

Reply:
Good point, I hadn't looked into existing tools.

Reply:
Looks like python has a library called 'cgi' with a [parse_header()](https://docs.python.org/2/library/cgi.html#cgi.parse_header) function which we can leverage. My only concern is that it handles duplicate parameters by keeping the last one (i.e. 'text/csv; label_size=1; label_size=2' produces a parameter dictionary with {'label_size': '2'}). Is this acceptable behavior?

Reply:
It also returns an empty dictionary if the parameter is only a name without a value (i.e. 'text/csv; label_size' -> `('text/csv', {})`), which would pass instead of erroring.

Reply:
Does the standard for MIME/HTTP headers mention rules for options in content-type? 
If not then IMO it's better to be permissive than restrictive for optional parameters.

Reply:
The protocol doesn't have any guidelines regarding these edge cases: https://www.w3.org/Protocols/rfc1341/4_Content-Type.html
I'd say it should be fine to just go with the default behavior provided by cgi.parse_header()